pkg/bindings/test/system_test.go

```diff
@@ -48,4 +56,90 @@ var _ = Describe("Podman system", func() {
 		cancelChan <- true
 		Expect(len(messages)).To(BeNumerically("==", 3))
 	})
+
+	It("podman system prune - pod,container stopped", func() {
+		// Start and stop a pod to enter in exited state.
+		_, err := pods.Start(bt.conn, newpod)
+		Expect(err).To(BeNil())
+		_, err = pods.Stop(bt.conn, newpod, nil)
+		Expect(err).To(BeNil())
+		// Start and stop a container to enter in exited state.
+		var name = "top"
+		_, err = bt.RunTopContainer(&name, &bindings.PFalse, nil)
+		Expect(err).To(BeNil())
+		err = containers.Stop(bt.conn, name, nil)
+		Expect(err).To(BeNil())
+
+		systemPruneResponse, err := system.Prune(bt.conn, &bindings.PTrue, &bindings.PFalse)
+		Expect(err).To(BeNil())
+		Expect(len(systemPruneResponse.PodPruneReport)).To(Equal(1))
+		Expect(len(systemPruneResponse.ContainerPruneReport.ID)).To(Equal(1))
+		Expect(len(systemPruneResponse.ImagePruneReport.Report.Id)).
+			To(BeNumerically(">", 0))
+		Expect(systemPruneResponse.ImagePruneReport.Report.Id).
+			To(ContainElement("docker.io/library/alpine:latest"))
+		Expect(len(systemPruneResponse.VolumePruneReport)).To(Equal(0))
+	})
+
+	It("podman system prune running alpine container", func() {
+		// Start and stop a pod to enter in exited state.
+		_, err := pods.Start(bt.conn, newpod)
+		Expect(err).To(BeNil())
+		_, err = pods.Stop(bt.conn, newpod, nil)
+		Expect(err).To(BeNil())
+		// Start and stop a container to enter in exited state.
+		var name = "top"
+		_, err = bt.RunTopContainer(&name, &bindings.PFalse, nil)
+		Expect(err).To(BeNil())
+		err = containers.Stop(bt.conn, name, nil)
+		Expect(err).To(BeNil())
+		// Start container and leave in running
+		var name2 = "top2"
+		_, err = bt.RunTopContainer(&name2, &bindings.PFalse, nil)
+		Expect(err).To(BeNil())
+
+		systemPruneResponse, err := system.Prune(bt.conn, &bindings.PTrue, &bindings.PFalse)
+		Expect(err).To(BeNil())
+		Expect(len(systemPruneResponse.PodPruneReport)).To(Equal(1))
+		Expect(len(systemPruneResponse.ContainerPruneReport.ID)).To(Equal(1))
+		Expect(len(systemPruneResponse.ImagePruneReport.Report.Id)).
+			To(BeNumerically(">", 0))
+		// Alpine image should not be pruned as used by running container
+		Expect(systemPruneResponse.ImagePruneReport.Report.Id).
+			ToNot(ContainElement("docker.io/library/alpine:latest"))
+		Expect(len(systemPruneResponse.VolumePruneReport)).To(Equal(0))
+	})
+
+	It("podman system prune running alpine container volume prune", func() {
```

Comment:
This test didn't find coding error at https://github.com/containers/libpod/pull/5907/files#diff-d20431f0bfae3cfadda9b39a8ab6b074R78

Reply:
Thank you for the catch there. Fixed the test to validate the use case https://github.com/containers/libpod/pull/5907/commits/0820e7c82abd1b84b3e81627f4ed396fb0f339bd#diff-745bc12747465f015915254b3f4917e1R101-R115